src/main/groovy/com/codebrig/arthur/observe/structure/literal/JavascriptLiteral.groovy

```diff
@@ -9,26 +9,44 @@ import com.codebrig.arthur.observe.structure.StructureLiteral
  * @version 0.4
  * @since 0.2
  * @author <a href="mailto:brandon.fergerson@codebrig.com">Brandon Fergerson</a>
+ * @author <a href="mailto:valpecaoco@gmail.com">Val Pecaoco</a>
  */
 class JavascriptLiteral extends StructureLiteral {
 
     @Override
     String getNodeLiteralAttribute(SourceNode node) {
         switch (Objects.requireNonNull(node).internalType) {
             case "NumericLiteral":
+                if (node.token.contains(".") || (node.token.isDouble() && node.token.toUpperCase().contains("E"))) {
```

Comment:
Same question, if we can tell it's a double (via `node.token.isDouble()`) do we need the further checks?

Reply:
Please see https://github.com/CodeBrig/Arthur/pull/79#discussion_r305654377 above